Foundation.hs

```diff
@@ -149,12 +151,17 @@ instance YesodAuth App where
 
 buildUser :: Creds m -> Maybe User
 buildUser (Creds csPlugin csIdent csExtra) =
-    User <$> lookup "first_name" csExtra
-         <*> lookup "last_name" csExtra
+    User <$> buildName csExtra
          <*> lookup "email" csExtra
          <*> pure csPlugin
          <*> pure csIdent
 
+buildName :: [(Text, Text)] -> Maybe Text
+buildName csExtra = do
+    firstName <- lookup "first_name" csExtra
+    lastName <- lookup "last_name" csExtra
+    return $ firstName <> " " <> lastName
```

Comment:
Thoughts on moving `buildUser`, `buildName`, and `repalceUser` into `Model.User`?